src/Vlingo.Common/Scheduler.cs

```diff
@@ -53,12 +79,13 @@ public virtual void Close()
             bool repeats)
         {
             var task = new SchedulerTask(scheduled, data, delayBefore, interval, repeats);
-            tasks.Add(task);
+            tasks.Push(task);
             return task;
         }
 
-        private class SchedulerTask : ICancellable
+        private class SchedulerTask : ICancellable, IDisposable
```

Comment:
I don't think the `IDisposabe` implementation here will be used. Because instances of this class are exposed as `ICancellable` through the `Scheduler` class. It is sufficient to implement `IDisposable` in the `Scheduler` and call `SchedulerTask.Cancel()` on Close/Dispose event.

Reply:
Oh yeah, this is the private class so semantics are different. As there is no risk to use it outside of the `Scheduler` we can be use that it won't be misused. I'll make some changes. Thanks

Reply:
Removed